envoyauth/request.go

```diff
@@ -160,6 +160,35 @@ func getParsedBody(logEntry *logrus.Entry, headers map[string]string, body strin
 			if !known {
 				return nil, false, nil
 			}
+		} else if strings.Contains(val, "application/x-www-form-urlencoded") {
+			if body == "" {
+				return nil, false, nil
+			}
+
+			if val, ok := headers["content-length"]; ok {
+				truncated, err := checkIfHTTPBodyTruncated(val, int64(len(body)))
+				if err != nil {
+					return nil, false, err
+				}
+				if truncated {
+					return nil, true, nil
+				}
+			}
+
+			parsed, err := url.ParseQuery(body)
+			if err != nil {
+				return nil, false, err
+			}
+
+			jsonBody, err := json.Marshal(parsed)
+			if err != nil {
+				return nil, false, err
+			}
+
+			err = util.UnmarshalJSON([]byte(jsonBody), &data)
+			if err != nil {
+				return nil, false, err
+			}
```

Comment:
💭 Do we need the roundtrip? `parsed` is `url.Values`, which is an alias of `map[string][]string`. So, how about

```go
data = interface{}(parsed)
```
or something similar...? 🤔 

I believe we might even be able to say `data, err = url.ParseQuery(body)`, but I haven't tried that.

Reply:
@srenatus I think you are right, we can just return the `parsed` data as a `url.Values` type and use that same type for object we are asserting with. The initial reason I went with converting to JSON and back was to not use the `url.Values` type in the tests but I think its fine! I pushed a fix in f76cbd6!

Reply:
> The initial reason I went with converting to JSON and back was to not use the url.Values type in the tests but I think its fine! 

Hm! I guess if we turn `url.Values` back into the native type, i.e. `data = parsed.(map[string][]string)`, we wouldn't have to do that, would we? What do you think?

Reply:
I poked at this for a bit but I think the only way to convert the `url.Values` type back to its native type is by looping over the values and copying them one by one into a new variable with the desired type, which might be better then doing thee marshal/unmarshal dance. 

Here's an example: https://play.golang.org/p/jZJ-J7bTUGk

Reply:
Following up, I actually ran a benchmark (just on my laptop) with the two options and it looks like looping through the values and copying it into another variable ended up being faster. Here's the [code](https://play.golang.org/p/ntlkHApdK65) (the playground won't work 😄 )

```
goos: darwin
goarch: amd64
pkg: example
cpu: Intel(R) Core(TM) i7-5650U CPU @ 2.20GHz
BenchmarkLoop-4          1691784               831.3 ns/op
BenchmarkMarshal-4        175098              6107 ns/op
PASS
ok      example 3.472s
```

Reply:
My bad. Could you try `data = map[string][]string(parsed)`, please? I'd suspect it to be quicker, too.

Thanks for bearing with me! 😅

Reply:
ah yeah🤦 , its a definitely bit faster and cleaner. Got a little side tracked on all the possibilities 👨‍🔬 . Fixed in 9314ee5!